coffeecard/CoffeeCard/Services/ProductService.cs

```diff
@@ -13,9 +15,26 @@ public ProductService(CoffeecardContext context)
             _context = context;
         }
 
-        public IEnumerable<Product> GetProducts()
+        public async Task<IEnumerable<Product>> GetPublicProducts()
         {
-            return _context.Products.AsEnumerable();
+            return await GetProducts(false);
+        }
+
+        public async Task<IEnumerable<Product>> GetProductsForUserAsync(User user)
```

Comment:
If we are more specific than just barista, these would need to be changed ofcourse